src/PdfSmith.BusinessLayer/Templating/HandlebarsTemplateEngine.cs

```diff
@@ -0,0 +1,122 @@
+        }
+    }
+
+    private static void RegisterGlobalHelpers()
+    {
+        var handlebarsInstance = Handlebars.Create();
+
+        // Register helper for formatting currency
+        Handlebars.RegisterHelper("formatCurrency", (context, arguments) =>
+        {
+            if (arguments.Length > 0 && decimal.TryParse(arguments[0].ToString(), out var value))
+            {
+                // Use the current culture for formatting
+                return value.ToString("C", CultureInfo.CurrentCulture);
+            }
+            return arguments.FirstOrDefault()?.ToString() ?? string.Empty;
+        });
+
+        // Register helper for formatting dates
+        Handlebars.RegisterHelper("formatDate", (context, arguments) =>
+        {
+            if (arguments.Length > 0)
+            {
+                var dateValue = arguments[0];
+                var format = arguments.Length > 1 ? arguments[1].ToString() : "yyyy-MM-dd";
+                
+                if (dateValue is DateTime dateTime)
+                {
+                    return dateTime.ToString(format, CultureInfo.CurrentCulture);
+                }
+                if (dateValue is DateTimeOffset dateTimeOffset)
+                {
+                    return dateTimeOffset.ToString(format, CultureInfo.CurrentCulture);
+                }
+                if (DateTime.TryParse(dateValue.ToString(), out var parsedDate))
+                {
+                    return parsedDate.ToString(format, CultureInfo.CurrentCulture);
+                }
+            }
+            return arguments.FirstOrDefault()?.ToString() ?? string.Empty;
+        });
+
+        // Register helper for accessing DateTime.Now with timezone support
+        Handlebars.RegisterHelper("dateTimeNow", (context, arguments) =>
+        {
+            return timeProvider?.GetLocalNow().DateTime ?? DateTime.Now;
+        });
+
+        // Register helper for accessing DateTimeOffset.Now with timezone support
+        Handlebars.RegisterHelper("dateTimeOffsetNow", (context, arguments) =>
+        {
+            return timeProvider?.GetLocalNow() ?? DateTimeOffset.Now;
+        });
+
+        // Register helper for mathematical operations
+        Handlebars.RegisterHelper("multiply", (context, arguments) =>
+        {
+            if (arguments.Length >= 2 && 
+                decimal.TryParse(arguments[0].ToString(), out var value1) &&
+                decimal.TryParse(arguments[1].ToString(), out var value2))
+            {
+                return value1 * value2;
+            }
+            return 0m;
+        });
+    }
+
+    [GeneratedRegex(@"(?<![\\w$])(?:@)?(?:System\.)?DateTime\.Now(?![\\w$])")]
+    private static partial Regex DateTimeNowRegex { get; }
+
+    [GeneratedRegex(@"(?<![\\w$])(?:@)?(?:System\.)?DateTimeOffset\.Now(?![\\w$])")]
+    private static partial Regex DateTimeOffsetNowRegex { get; }
```

Comment:
Does Handlebars allow to use C# code in the template, like Razor? If no, this code that checks for C# code in the template is useless.